pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -879,6 +879,15 @@ func pvDeleted(obj interface{}) {
 	}
 }
 
+// GetAllK8sVolumes returns list of volumes in a bound state
+func (c *K8sOrchestrator) GetAllK8sVolumes() []string {
+	volumeIDs := make([]string, 0)
+	for volumeID := range c.volumeIDToNameMap.items {
```

Comment:
This map contains only CSI volumes, and does not contain VCP-CSI migrated volumes. It is because of https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/008b6925359f8633e8ea8129b3a27298189eab22/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go#L806 & https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/008b6925359f8633e8ea8129b3a27298189eab22/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go#L841, where it checks if the volume is of type CSI. For VCP - CSI migrated volume pv.Spec.CSI will be nil. 

In a cluster if we have 100 vcp-csi migrated volumes, cns query will return all the volumes including the migrated ones.
Hence, when ListVolumes does a diff of k8s volumes and cns volumes, cns volumes will keep returning 100 volumes more than k8s volumes which could be more than the threshold.